packages/sdk/src/shared/index.ts

```diff
@@ -191,28 +268,111 @@ export type Account = {
   credentials?: Record<string, string>;
 };
 
-export type ComponentReloadPropsOpts = {
-  userId: string;
-  componentId: string;
-  configuredProps: any;  // eslint-disable-line @typescript-eslint/no-explicit-any
+/**
+ * The request options for reconfiguring a component's props when dealing with
+ * dynamic props.
+ */
+export type ReloadComponentPropsOpts = ExternalUserId & {
+  /**
+   * The ID of the component you're configuring. This is the key that uniquely
+   * identifies the component.
+   */
+  componentId: string | ComponentId;
+
+  /**
+   * The props that have already been configured for the component. This is a
+   * JSON-serializable object with the prop names as keys and the configured
+   * values as values.
+   */
+  configuredProps: ConfiguredProps<ConfigurableProps>;
+
+  /**
+   * The ID of the last prop reconfiguration (or none when reconfiguring the
+   * props for the first time).
+   */
   dynamicPropsId?: string;
 };
 
-export type ComponentConfigureOpts = {
-  userId: string;
-  componentId: string;
+/**
+ * @deprecated Use `ReloadComponentPropsOpts` instead.
+ */
+export type ComponentReloadPropsOpts = ReloadComponentPropsOpts;
+
+/**
+ * The request options for configuring a component's prop.
+ */
+export type ConfigureComponentOpts = ExternalUserId & {
+  /**
+   * The ID of the component you're configuring. This is the key that uniquely
+   * identifies the component.
+   */
+  componentId: string | ComponentId;
+
+  /**
+   * The name of the prop you're configuring.
+   */
   propName: string;
-  configuredProps: any;  // eslint-disable-line @typescript-eslint/no-explicit-any
+
+  /**
+   * The props that have already been configured for the component. This is a
+   * JSON-serializable object with the prop names as keys and the configured
+   * values as values.
+   */
+  configuredProps: ConfiguredProps<ConfigurableProps>;
+
+  /**
+   * The ID of the last prop reconfiguration (if any).
+   */
   dynamicPropsId?: string;
+
   query?: string;
 };
 
-export type GetComponentOpts = RelationOpts & {
+/**
+ * @deprecated Use `ConfigureComponentOpts` instead.
+ */
+export type ComponentConfigureOpts = ConfigureComponentOpts;
+
+/**
+ * The request options for retrieving a list of components.
+ */
+export type GetComponentsOpts = RelationOpts & {
+  /**
+   * A search query to filter the components.
+   */
   q?: string;
+
+  /**
+   * The ID or name slug of the app to filter the components.
+   */
   app?: string;
-  componentType?: "trigger" | "action";
+
+  /**
+   * The type of component to filter (either "trigger" or "action").
+   */
+  componentType?: ComponentType;
+};
+
+/**
+ * @deprecated Use `GetComponentsOpts` instead.
+ */
+export type GetComponentOpts = GetComponentsOpts;
+
+export type ComponentId = {
```

Comment:
Should this be `ComponentKey`? Will this always be a `key` string or potentially a hash ID?